tests/suites/test_suite_pk.function

```diff
@@ -526,9 +527,23 @@ void pk_rsa_verify_ext_test_vec( data_t * message_str, int digest,
         pss_opts.expected_salt_len = salt_len;
     }
 
-    TEST_ASSERT( mbedtls_pk_verify_ext( pk_type, options, &pk,
-                                digest, hash_result, hash_len,
-                                result_str->x, sig_len ) == result );
+    ret = mbedtls_pk_verify_ext( pk_type, options, &pk,
+                                 digest, hash_result, hash_len,
+                                 result_str->x, sig_len );
+
+    /* Mbed TLS distinguishes "invalid padding" from "valid padding but
+     * the rest of the signature is invalid". This has little use in
+     * practice and PSA doesn't report this distinction.
+     * In this case, PSA returns PSA_ERROR_INVALID_SIGNATURE translated
+     * to MBEDTLS_ERR_RSA_VERIFY_FAILED
+     */
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if( result == MBEDTLS_ERR_RSA_INVALID_PADDING &&
+        ret == MBEDTLS_ERR_RSA_VERIFY_FAILED )
```

Comment:
I think this line should be removed, the `TEST_EQUAL()` below is what we want - it must always be called, so that it can report an error is `ret` is not as expected.

Reply:
Or we could even do:
```c
#if defined(MBEDTLS_USE_PSA_CRYPTO)
    if( result == MBEDTLS_ERR_RSA_INVALID_PADDING )
        result = MBEDTLS_ERR_RSA_VERIFY_FAILED;
#endif
    TEST_EQUAL( ret, result );

Reply:
I understand, but actually this fails on:
```
Verify ext RSA #7 (PKCS1 v2.1, wrong MGF1 alg != MSG hash alg) .... FAILED
  ret == result
  at line 544, suites/test_suite_pk.function
  lhs = 0xffffffffffffbf00 = -16640
  rhs = 0xffffffffffffbc80 = -17280
```

because `mbedtls_pk_verify_ext` returns `MBEDTLS_ERR_RSA_VERIFY_FAILED` on wrong hash alg.

Reply:
Ok, I re-read the surrounding code, and it seems like I missed the `else` last time, which ensures that one of the two `TEST_EQUAL()` is always executed, so that's fine.